ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +305,591 @@ def test_ipa_user_s4u2self_pac(self):
+        msg = (
+            'adding SIDs to existing users and groups\n'
+            'This step may take considerable amount of time, please wait..'
+        )
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "--add-sids",
+             "-U"], raiseonerr=False
+        )
+        assert msg in result.stdout_text
+
+    def test_cldap_responder_doesnot_hang_for_domain_discovery(self):
+        """
+        This testcase checks that cldap responder doesnot hang
+        for domain discovery.
+        https://pagure.io/freeipa/issue/3639
+        """
+        version = tasks.get_openldap_client_version(self.master)
+        if version >= '2.6':
+            pytest.skip('bz2167328')
+        base_dn = str(self.master.domain.basedn)
+        srch_filter = "(&(DnsDomain='{}')(NtVer=\06\00\00\00)" \
+                      "(AAC=\00\00\00\00))".format(base_dn)
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        result = self.master.run_command(
+            ["ldapsearch", "-LL", "-H",
+             "cldap://{}".format(self.master.hostname),
+             "-b", "", "-s", "base", base_dn, srch_filter]
+        )
+        assert result.returncode == 0
+        assert 'dn: \n netlogon::*' in result.stdout_text
+
+    def test_user_connects_smb_share_if_locked_specific_group(self):
+        """
+        This testcase checks that with ipa-adtrust-installed and
+        the sidgen task registered and run, users cannot connect
+        a share in the local samba server if it is locked to specific
+        groups via "valid users", even if the user is a member of one
+        of those groups.
+        https://pagure.io/freeipa/issue/4234
+        """
+        msg = "tree connect failed: NT_STATUS_ACCESS_DENIED"
+        self.master.run_command(
+            ["ipa-adtrust-install", "-a",
+             self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        self.master.run_command(["mkdir", "/freeipa4234"])
+        self.master.run_command(
+            ["chcon", "-t", "samba_share_t",
+             "/freeipa4234"])
+        self.master.run_command(
+            ["setfacl", "-m", "g:admins:rwx",
+             "/freeipa4234"])
+        self.master.run_command(
+            ["net", "conf", "setparm", "share",
+             "comment", "Test Share"])
+        self.master.run_command(
+            ["net", "conf", "setparm", "share",
+             "read only", "no"])
+        self.master.run_command(
+            ["net", "conf", "setparm", "share",
+             "path", "/freeipa4234"])
+        self.master.run_command(["touch", "before"])
+        self.master.run_command(["touch", "after"])
+        self.master.run_command(
+            ["echo", "put", "before", "|",
+             "smbclient", "-k",
```

Comment:
smbclient -k is deprecated, you should use smbclient --use-kerberos=desired (see https://pagure.io/freeipa/issue/8926)
Instead of using a pipe, you can use smbclient -c "put before"